src/lib/offline-api/index.ts

```diff
@@ -321,6 +359,10 @@ class OfflineAPI {
     return this.requestBatches
   }
 
+  public recordTransformError (error) {
```

Comment:
Depending on the use-case this might turn async later as well - or we have to always store errors in-memory